src/main/kotlin/minesweeper/domain/MineMap.kt

```diff
@@ -2,29 +2,61 @@ package minesweeper.domain
 
 class MineMap(private val width: Width, private val height: Height) {
 
-    val mineMap = List(width.getWidth()) {
-        TileRow(List(height.getHeight()) { Tile(TileType.PLAIN) })
+    val mineMap = List(height.height) {
+        TileRow(MutableList(width.width) { PlainTile() })
     }
 
     fun makeMine(mineCount: MineCount) {
-        require(width.getWidth() * height.getHeight() > mineCount.getMineCount()) {
+        require(width.width * height.height > mineCount.count) {
             "지뢰는 넓이*너비의 수보다 작아야한다"
         }
         var makeCount = 0
         do {
             makeCount = setMineToTile(makeCount)
-        } while (makeCount != mineCount.getMineCount())
+        } while (makeCount != mineCount.count)
     }
 
     private fun setMineToTile(makeCount: Int): Int {
-        val rangeX = width.getWidth()
-        val rangeY = height.getHeight()
-        val (x, y) = RandomPosition.of(rangeX, rangeY)
-        val tile = mineMap[x][y]
+        val position = RandomPosition.of(height.height, width.width)
+        val randomHeight = position.randomHeight
+        val randomWidth = position.randomWidth
+        val positionX = randomHeight.position
+        val positionY = randomWidth.position
+        val tile = mineMap[positionX][positionY]
         if (!tile.isMine()) {
-            tile.setMine()
+            mineMap[positionX][positionY] = MineTile()
+            checkMine(positionX, positionY)
             return makeCount.inc()
         }
         return makeCount
     }
+
+    private fun checkMine(x: Int, y: Int) {
+        for (column in x.dec()..x.inc()) {
+            checkMineRow(column, y)
+        }
+    }
+
+    private fun checkMineRow(column: Int, y: Int) {
+        if (column < 0 || column >= mineMap.size) {
+            return
+        }
+        for (row in y.dec()..y.inc()) {
+            setMineCount(column, row)
+        }
+    }
+
+    private fun setMineCount(column: Int, row: Int) {
+        if (row < 0 || row >= mineMap[column].size) {
+            return
+        }
+        checkNearMineCount(mineMap[column][row])
+    }
+
+    private fun checkNearMineCount(tile: Tile) {
+        when (tile) {
+            is MineTile -> return
+            is PlainTile -> tile.fineNearMine()
+        }
```

Comment:
이런 경우라면 아래 코드가 더 간결하지 않을까요? 
```suggestion
        if (tile is PlainTile) { tile.fineNearMine() }
```

Reply:
수정해보았습니다.